### PR TITLE
refactor(client): remove crosstab.

### LINKED ISSUE #3415: Discuss: remove crosstab.js dependency
Ref #3414 

crosstab was useful when we first started FxA because it allows straight forward communication between two tabs within the same browser. Firefox implemented [BroadcastChannel](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) which serves the same purpose, without the use of a library.

crosstab's communication mechanism between two tabs relies on shuttling data via localStorage. If there are problems accessing localStorage [1][2][3][4] within crosstab, our startup process fails and the user can see a [blank screen](https://github.com/mozilla/fxa-content-server/issues/3414#issuecomment-173911165). If crosstab is able to read from but not write to localStorage, then an error is sent to Sentry [every 3 seconds](https://github.com/mozilla/fxa-content-server/issues/3414#issue-128142184). Sentry reports > 5M instances of "main in setLocalStorageItem" types of errors, making it impossible to determine whether a large number of users see this problem or whether it's a small number of users who submit a large number of reports.

Removing crosstab will affect users of Firefox < 38 or users who signup to FxA from other browsers. The password reset flow from about:accounts will require users to return to the original tab to re-enter their credentials. Tab signin/signout state will not stay in sync. crosstab already disables itself for mobile users, so Fennec should not be affected. Removing crosstab will make for a marginally worse UX for a subset of users, but will result in less code to maintain and fewer code paths to worry about.

[1] - https://github.com/mozilla/fxa-content-server/issues/2969
[2] - https://github.com/mozilla/fxa-content-server/issues/2913
[3] - https://github.com/mozilla/fxa-content-server/issues/3414
[4] - https://bugzilla.mozilla.org/show_bug.cgi?id=1240238

cc @vladikoff, @ckarlof, @rfk, @philbooth, @zaach, @vbudhram, @ryanfeeley, @jrgm

### PR DESCRIPTION
**Pull request by shane-tomlinson**
All new Firefox versions support BroadcastChannels.

fixes #3415 

I've had this code lying around for months. `crosstab` has been a source of frustration since we added it.

We originally added crosstab to communicate between tabs and [BroadcastChannel](https://developer.mozilla.org/en-US/docs/Web/API/Broadcast_Channel_API) was not yet a thing.  BroadcastChannel is supported in Firefox >= 38.

In addition, Fx < 39 did not have WebChannel support. This meant FxA could *only* send `login` messages containing Sync creds from FxA to the browser from within about:accounts. This was problematic for password reset where users enter their email in one tab, and their new password in a 2nd tab. cross-tab communication was added to make this flow possible:

1. User opens about:accounts, realizes they forgot their password, and initiates a password reset.
2. User verifies in the same browser in a second tab. User enters new password into 2nd tab.
3. Either the new password or Sync creds need to be sent from the 2nd tab to the `about:accounts` tab where they can then be sent to the browser.

With WebChannels the 2nd tab can send the Sync creds to the browser from *any* tab, without the need to send information to the first tab.

Who will this PR break?

* Users of Firefox Desktop < 39 who reset their password from about:accounts and verify in the same browser. User will need to return to the original tab where they should be able to sign in.
* Users of Firefox Desktop < 39 who sign in or sign out, if more than two tabs are kept open, "other" tabs will not be kept in sync.

### CODE REVIEW

**Comment by rfk** [comment]
> Who will this PR break?

This breakage seems fine to me; re:dash should be able to tell us approximate numbers on those two categories of user.

**Comment by shane-tomlinson** [comment]
[Number of Fx Desktop users on Fx < 38 who trigger an `account.reset` activity event](https://sql.telemetry.mozilla.org/queries/1302):

screen shot 2016-09-28 at 11 04 42

**Comment by vladikoff** [comment]
needs rebase 🎈

**Comment by shane-tomlinson** [comment]
@rfk pointed out that my query needs to multiply by 10 AND that I'm using the old `events` table instead of the new `activity_events` table. New query needed!

**Comment by shane-tomlinson** [comment]
Next steps: rebase and get better numbers.

**Comment by shane-tomlinson** [comment]
Some [new data](https://sql.telemetry.mozilla.org/queries/1302/source#table):

screen shot 2016-10-19 at 12 02 07

**Comment by rfk** [comment]
I support degrading the experience for this small number of users, in the interests of reducing complexity in our code.

File: app/scripts/lib/channels/inter-tab.js [hunk 1]
```diff
@@ -4,51 +4,41 @@
...
- * communicate: BroadcastChannelAdapter or LocalStorageAdapter.
+ * tabs of the same browser, using BroadcastChannels
+ * to communicate.
  *
- * The preferred adapter is the BroadcastChannelAdapter. It uses
- * the BroadcastChannel API to communicate between browser tabs. See
- * https://developer.mozilla.org/docs/Web/API/Broadcast_Channel_API.
- * The BroadcastChannelAdapter is the prefered way to communicate
- * because it does not write any sensitive data to disk.
+ * See https://developer.mozilla.org/docs/Web/API/Broadcast_Channel_API.
```
Comment (vladikoff) [inline app/scripts/lib/channels/inter-tab.js]: 👍

File: app/scripts/lib/channels/inter-tab.js [hunk 2]
```diff
@@ -4,51 +4,41 @@
...
  *
- * LocalStorageAdapter uses localStorage and storage events to communicate
- * between tabs. LocalStorageAdapter is used for legacy browsers without
- * BroadcastChannel API support. Any data sent over this channel
- * can be written to disk, so care must be taken to ensure all
- * sensitive data is erased as soon as it is consumed by the receiving tab.
+ * If BroadcastChannel is not supported, no inter-tab communication occurs.
```
Comment (vladikoff) [inline app/scripts/lib/channels/inter-tab.js]: nit: maybe add a note about supported Firefox versions here or a link to this PR?

**Comment by vladikoff** [comment]
@shane-tomlinson I think you left `crosstab` in `require_config.js`:

image

and 

```
shim: {
    backbone: {
      deps: [
        'underscore',
        'jquery'
      ],
      exports: 'Backbone'
    },
    crosstab: {
      exports: 'crosstab'
    },
```

^ Need to remove that and maybe adjust the `nit` comment above. r+ otherwise

**Comment by vladikoff** [comment]
Unrelated fail due to `@gmail` addresses https://github.com/mozilla/fxa-content-server/issues/4318

**Pull request merged**